app/scripts/directives/blocks.js

```diff
@@ -89,26 +89,40 @@ angular
       templateUrl: radioQuestionTemplate,
       restrict: 'E',
       controller: function ($scope) {
-        $scope.$watch('answer.value', function () {
-          if (angular.isDefined($scope.answer)) {
-            //check if answer is not in current choices
-            if (
-              !_.includes(
-                _.map($scope.block.content.choices, 'value'),
-                $scope.answer.value,
-              )
-            ) {
-              $scope.otherAnswer = $scope.answer.value;
-            }
-          }
-        });
+        $scope.otherSentinel = '__other__';
 
-        $scope.selectOtherAnswer = function () {
-          if (!$scope.answer) {
-            return;
+        $scope.selectOtherAnswer = () => {
+          if ($scope.otherAnswer) {
+            $scope.selectedAnswer = $scope.otherSentinel;
           }
-          $scope.answer.value = $scope.otherAnswer;
         };
+
+        $scope.clearAnswer = () => {
+          $scope.selectedAnswer = '';
```

Comment:
I noticed this line isn't covered by tests. Did you get to test the case where you type in an "Other" answer, submit the registration, edit the registration, clear the Other answer, and submit again? I think that was the edge case I didn't get working before, basically I could submit an empty other answer.

Reply:
I can add tests to cover the missing lines, but the Other input is required when the other option is selected, so it doesn't let you clear the other answer and submit.